### ISSUE TITLE
Error occured when removing duplicates

### ISSUE THREAD

**Issue by YihongT** [understanding-search/maze-dataset#42]
I encountered an error while generating the mazes: 'Error removing duplicates: generation meta is not present in a maze, cannot collect generation meta.'

```python
self.dataset = MazeDataset.from_config(
    self.cfg,
    load_local=self.args.load,
    save_local=self.args.save,
    local_base_path=self.maze_path,
    verbose=True,
    zanj=ZANJ(external_list_threshold=1000)
)
logging.info(f'Generated {len(self.dataset)} mazes')

try:
    self.dataset = self.dataset.filter_by.remove_duplicates(
        minimum_difference_connection_list=0, 
        minimum_difference_solution=1
    )
    logging.info(f'After removing duplicates, we have {len(self.dataset)} mazes')
except Exception as e:
    logging.error(f'Error removing duplicates: {e}')
```

I am using the DFS with percolation algorithm to generate 300 mazes with a scale of 4.
```python
maze_ctor = LatticeMazeGenerators.gen_dfs_percolation
maze_ctor_kwargs = dict(p=0.4)
```

Why am I encountering this error and how can I fix it?

**Comment by mivanit** [comment]
@YihongT I'm working on this in #43, attempting to reproduce.

It's also worth noting that unless you are dealing with very small maze sizes or very large datasets, duplicates are exceptionally rare, and the de-duplication filter often has no effect.

**Comment by mivanit** [comment]
A quick and hacky fix that should fix the issue, since it appears to be only prevalent when using minimal serialization:
```python
from maze_dataset import set_serialize_minimal_threshold
set_serialize_minimal_threshold(int(1e10)) # or any number larger than your dataset size
```

working on a more robust fix now, and will push an update to pypi

**Comment by mivanit** [comment]
should be fixed with [0.5.5](https://pypi.org/project/maze-dataset/0.5.5/)
